### PR TITLE
Add the ability to use precalculated charge densities in Bader method

### PR DESCRIPTION
**Pull request by mscho527**
_Related Issue: #886 
Depends on: #900_ 

This PR adds the ability to prepare `Volume` objects from pre-calculated `.cube` files and allows this pre-calculated charge densities to be used to initiate Bader charges calculations. A test that does a round-trip conversion from `Volume` to cube file to `Volume` are included to verify the functionalities.

### CODE REVIEW

File: cclib/method/volume.py [hunk 1]
```diff
@@ -361,3 +365,57 @@ def electrondensity(ccdata, volume, mocoeffslist):
...
+            skiplines += 1
+        
+        # Line 10+ (or 7+ if atomic coordinates data are not present)
+        tmp = []
+        for line in lines[skiplines:]:
+            tmp.extend(line.split())
+        tmp = numpy.asanyarray(tmp, dtype = float)
+    
+    # Initialize volume object
+    vol = Volume((convert(originx - 0.5 * ngridx * spacingx), convert(originy - 0.5 * ngridy * spacingy), convert(originz - 0.5 * ngridz * spacingz)), (convert(originx + 0.5 * ngridx * spacingx - 0.5 * spacingx), convert(originy + 0.5 * ngridy * spacingy - 0.5 * spacingy), convert(originz + 0.5 * ngridz * spacingz - 0.5 * spacingz)), (convert(spacingx), convert(spacingy), convert(spacingz)))
```
Comment (berquist) [inline cclib/method/volume.py:413]: Can you run Black on this?

Comment (mscho527) [reply cclib/method/volume.py]: Yes -- I have just run black for this code. Thanks.

**Comment by shivupa** [comment]
Can you rebase this on master?

File: cclib/method/bader.py [hunk 2]
```diff
@@ -74,21 +74,28 @@ def calculate(self, indices=None, fupdate=0.05):
...
            """
 
-        # First obtain charge densities on the grid
-        if len(self.data.mocoeffs) == 1:
-            self.chgdensity = electrondensity_spin(
-                self.data, self.volume, [self.data.mocoeffs[0][: self.data.homos[0]]]
-            )
-            self.chgdensity.data *= 2
+        # Obtain charge densities on the grid if it does not contain one.
+        if self.volume.data.sum() == 0.0:
```
Comment (berquist) [inline cclib/method/bader.py:78]: Do you think there's a better way to check this?

Comment (mscho527) [reply cclib/method/bader.py]: Yes -- `numpy.any` should do the job. Thanks for pointing this out!

File: cclib/method/volume.py [hunk 3]
```diff
@@ -365,3 +367,68 @@ def electrondensity(ccdata, volume, mocoeffslist):
...
+    Input:
+        filepath -- path to the cube file
+        
+    Output:
+        vol -- Volume object filled with data from cube file
+    """
+
+    # Cube files are written in bohrs
+    def convert(x):
+        return convertor(x, "bohr", "Angstrom")
```
Comment (shivupa) [inline cclib/method/volume.py:385]: `covertor(x, 'bohr', 'Angstrom')` isn't much longer and is more explicit when used. could this be changed back

Comment (mscho527) [reply cclib/method/volume.py]: I've modified this into a full call. Thanks!

**Pull request merged**